api/controllers/user.js

```diff
@@ -206,6 +208,46 @@ async function appleLogin(req, accessToken, refreshToken, idToken, profile, done
   return passportLogin(ip, 'apple', accessToken, refreshToken, appleProfile, done);
 }
 
+async function googleIdTokenLogin(req, res) {
+  const client = new OAuth2Client();
+  const id_token = req.query.id_token;
+  async function verify() {
+    const ticket = await client.verifyIdToken({
+      idToken: id_token,
+      audience: [process.env.GOOGLE_FIREBASE_SIGN_IN_APP_ID, process.env.GOOGLE_FIREBASE_WEB_CLIENT_SIGN_IN]
+      // Or, if multiple clients access the backend:
+      //[CLIENT_ID_1, CLIENT_ID_2, CLIENT_ID_3]
+    });
+    return ticket.getPayload();
+  }
+  try {
+    const profile = await verify();
+    const googleProfile = {
+      id: profile.sub,
+      accessToken: id_token,
+      gender: null,
+      displayName: profile.name,
+      name: profile?.given_name,
+      lastname: profile?.family_name,
+      email: profile.email,
+      emails: [{ value: profile.email }],
+      photos: [{ value: profile.picture }]
+    };
+    await googleLogin(req, id_token, null, googleProfile, (error, response) => {
+      if (error) {
+        console.error(err);
+        res.redirect('/');
```

Comment:
Why do you use a redirect here? I think we can directly use  `res.status(404).json({message: "Something went wrong on Google Id Token login"});`